### PR TITLE
Adding initial docs for the new import/export

### PR DESCRIPTION
**Pull request by parthaa**
Cherry-pick into:

* [ ] Foreman 2.3 (Satellite 6.9)
* [ ] Foreman 2.1 (Satellite 6.8)

### CODE REVIEW

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 1]
```diff
@@ -37,66 +37,117 @@ The exported archive file contains the following data:
...
+---|----------|---------|-------------|----------------------
+
 ----
 
-. Export the version of a Content View.
-Specify the directory where to store the export with the `--export-dir` option and the ID of the Content View version that you export with the `--id` option.
-The `pulp_export_destination` setting does not work for this procedure.
-+
+.Export the version of a Content View.
+Get the version number of desired version. In the case below we are interested in fully exporting version `1.0`.
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:69]:
```suggestion
Get the version number of desired version. In the case below we are interested in fully exporting version `1.0`.
```
```suggestion
Get the version number of desired version. The following example targets version `1.0` for export.
```

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 2]
```diff
@@ -37,66 +37,117 @@ The exported archive file contains the following data:
...
+
 [options="nowrap" subs="+quotes"]
 ----
-# hammer content-view version export --export-dir _export_directory_ \
---id _content_view_version_ID_
+# hammer content-export complete version --content-view=view --version=1.0 --organization=export-21527
+[.....................................................................................................................................................................................................................................] [100%]
+Generated /var/lib/pulp/exports/export-21527/view/1.0/2021-02-25T18-59-26-00-00/metadata.json
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:75]:
```suggestion

```
We don't expose the output in docs unless it is complex and vital to work with for the next step.

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 3]
```diff
@@ -37,66 +37,117 @@ The exported archive file contains the following data:
...
 +
 [options="nowrap" subs="+quotes"]
 ----
-# ls _export_directory_
-_export-1.tar_
+# ls -lh /var/lib/pulp/exports/export-21527/view/1.0/2021-02-25T18-59-26-00-00/
+total 49M
+-rw-r--r--. 1 pulp pulp 49M Feb 25 18:59 export-f8c5e34a-bc16-4ba8-8e87-a2810e453d17-20210225_1859.tar.gz
+-rw-r--r--. 1 pulp pulp 333 Feb 25 18:59 export-f8c5e34a-bc16-4ba8-8e87-a2810e453d17-20210225_1859-toc.json
+-rw-r--r--. 1 root root 360 Feb 25 18:59 metadata.json
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:86]:
```suggestion

```

Remove output

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 4]
```diff
@@ -37,66 +37,117 @@ The exported archive file contains the following data:
...
+
+.Export with chunking
+
+In many cases the exported archive content may be several gigabytes in size. We may want to split it smaller sizes or chunks. We can use the `--chunk-size-mb` flag directly in the export command to handle this. In the example below we specify `--chunk-size-mb=10` to split the archives in `10 MB` chunks.
+
+[options="nowrap" subs="+quotes"]
+----
+# hammer content-export complete version --content-view=view --version=1.0 --organization=export-21527 --chunk-size-mb=10
+[.....................................................................................................................................................................................................................................] [100%]
+Generated /var/lib/pulp/exports/export-21527/view/1.0/2021-02-25T21-15-22-00-00/metadata.json
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:100]:
```suggestion

```
Remove output

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 5]
```diff
@@ -37,66 +37,117 @@ The exported archive file contains the following data:
...
+
+# ls -lh  /var/lib/pulp/exports/export-21527/view/1.0/2021-02-25T21-15-22-00-00/
+total 49M
+-rw-r--r--. 1 pulp pulp  10M Feb 25 21:15 export-fb548099-7931-4f16-a7b9-7cb47b885037-20210225_2115.tar.gz.0000
+-rw-r--r--. 1 pulp pulp  10M Feb 25 21:15 export-fb548099-7931-4f16-a7b9-7cb47b885037-20210225_2115.tar.gz.0001
+-rw-r--r--. 1 pulp pulp  10M Feb 25 21:15 export-fb548099-7931-4f16-a7b9-7cb47b885037-20210225_2115.tar.gz.0002
+-rw-r--r--. 1 pulp pulp  10M Feb 25 21:15 export-fb548099-7931-4f16-a7b9-7cb47b885037-20210225_2115.tar.gz.0003
+-rw-r--r--. 1 pulp pulp 8.4M Feb 25 21:15 export-fb548099-7931-4f16-a7b9-7cb47b885037-20210225_2115.tar.gz.0004
+-rw-r--r--. 1 pulp pulp  909 Feb 25 21:15 export-fb548099-7931-4f16-a7b9-7cb47b885037-20210225_2115-toc.json
+-rw-r--r--. 1 root root  360 Feb 25 21:15 metadata.json
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:110]:
```suggestion
```

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 6]
```diff
@@ -37,66 +37,117 @@ The exported archive file contains the following data:
...
+.Incremental Export
+
+Exporting complete versions can be a very expensive operation in terms of space and resources. Content view versions that have multiple RHEL trees may occupy several gigabytes of the space on {ProjectServer}.
+
+{ProjectServer} offers *Incremental Export* to help with this scenario. *Incremental Export* exports only things that changed from the previously exported version. These would be typically smaller than the full exports. In the example below we will incrementally export version `2.0`. Since we have already exported version `1.0` in earlier calls this would only include things that changed between version `1.0` and `2.0`
+
+----
+# hammer content-export incremental version --content-view=view --version=2.0 --organization=export-21527
+[............................................................................................................................................................................................................] [100%]
+Generated /var/lib/pulp/exports/export-21527/view/2.0/2021-02-25T21-45-34-00-00/metadata.json
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:123]:
```suggestion

```
Remove output

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 7]
```diff
@@ -37,66 +37,117 @@ The exported archive file contains the following data:
...
+
+# ls -lh /var/lib/pulp/exports/export-21527/view/2.0/2021-02-25T21-45-34-00-00/
+total 20M
+-rw-r--r--. 1 pulp pulp 20M Feb 25 21:45 export-ab5b4c58-51eb-49e4-8484-9ff2ea50c122-20210225_2145.tar.gz
+-rw-r--r--. 1 pulp pulp 333 Feb 25 21:45 export-ab5b4c58-51eb-49e4-8484-9ff2ea50c122-20210225_2145-toc.json
+-rw-r--r--. 1 root root 482 Feb 25 21:45 metadata.json
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:129]:
```suggestion

```

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 8]
```diff
@@ -37,66 +37,117 @@ The exported archive file contains the following data:
...
+----
+
+=== Examining the exports
+
+You can query on the exports that you previously have created via the `hammer content-export list` command.
+
+----
+hammer content-export list --organization=export-21527
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:137]: This isn't a procedure so it shouldn't have a heading, please add this command to the end of the exporting procedure.

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 9]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
 
 To import a Content View, ensure that the {ProjectServer} where you want to import meets the following conditions:
 
-ifeval::["{build}" == "satellite"]
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:109]: This looks to me to still be relevant for disconnected scenarios. How else will they manage?

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 10]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
+# chown -R pulp:pulp /var/lib/pulp/imports/2021-02-25T21-15-22-00-00/
+# ls -lh  /var/lib/pulp/imports/2021-02-25T21-15-22-00-00/
+total 49M
+-rw-r--r--. 1 pulp pulp  10M Mar  1 22:19 export-fb548099-7931-4f16-a7b9-7cb47b885037-20210225_2115.tar.gz.0000
+-rw-r--r--. 1 pulp pulp  10M Mar  1 22:19 export-fb548099-7931-4f16-a7b9-7cb47b885037-20210225_2115.tar.gz.0001
+-rw-r--r--. 1 pulp pulp  10M Mar  1 22:19 export-fb548099-7931-4f16-a7b9-7cb47b885037-20210225_2115.tar.gz.0002
+-rw-r--r--. 1 pulp pulp  10M Mar  1 22:19 export-fb548099-7931-4f16-a7b9-7cb47b885037-20210225_2115.tar.gz.0003
+-rw-r--r--. 1 pulp pulp 8.4M Mar  1 22:19 export-fb548099-7931-4f16-a7b9-7cb47b885037-20210225_2115.tar.gz.0004
+-rw-r--r--. 1 pulp pulp  909 Mar  1 22:19 export-fb548099-7931-4f16-a7b9-7cb47b885037-20210225_2115-toc.json
+-rw-r--r--. 1 pulp pulp  360 Mar  1 22:19 metadata.json
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:180]:
```suggestion

```
Remove output

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 11]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
 
-. Copy the archived file with the exported Content View version to the `/var/lib/pulp/katello-export` directory on the {ProjectServer} where you want to import.
-. On the {ProjectServer} where you want to import, create a Content View with the same name and label as the exported Content View.
+. Copy the archived file with the exported Content View version to the `/var/lib/pulp/imports` directory on the {ProjectServer} where you want to import.
+. Set the permission of the archive files to `pulp:pulp`.
++
+[subs="+quotes"]
+----
+# chown -R pulp:pulp /var/lib/pulp/imports/2021-02-25T21-15-22-00-00/
+# ls -lh  /var/lib/pulp/imports/2021-02-25T21-15-22-00-00/
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:172]:
```suggestion
. Set the SELinux permission of the archive files to `pulp:pulp`.
+
[subs="+quotes"]
----
# chown -R pulp:pulp /var/lib/pulp/imports/2021-02-25T21-15-22-00-00/
----
+
. Verify that the SELinux permission change occurs: 
+
[subs="+quotes"]
----
# ls -lh  /var/lib/pulp/imports/2021-02-25T21-15-22-00-00/
```

One command per step.

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 12]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
+
+----
++
+. On the {ProjectServer} where you want to import, create a Content View with the same name and label as the exported Content View with the `Import Only` option set.
 For more information, see xref:Managing_Content_Views-Creating_a_Simple_Content_View_CLI[].
 . Ensure that you enable the repositories that the Products in the exported Content View version include.
 For more information, see xref:Importing_Content-Selecting_Red_Hat_Repositories_to_Synchronize[].
 . In the {Project} web UI, navigate to *Content* > *Products*, click the *Yum content* tab and add the same `Yum` content that the exported Content View version includes.
-. Until https://bugzilla.redhat.com/show_bug.cgi?id=1745081[BZ#1745081] is resolved, navigate to the `/var/lib/pulp/katello-export` directory:
+. Identify the Content View that you wish to import
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:189]:
```suggestion
. Identify the Content View that you want to import
```

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 13]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
 +
 [subs="+quotes"]
 ----
-# hammer content-view version import \
---export-tar /var/lib/pulp/katello-export/_exported_CV_archive_.tar \
---organization-id _Your_Organization_ID_
+# hammer content-import version --content-view=view \
+                                --organization=import-20639 \
+                                --path=/var/lib/pulp/imports/2021-02-25T21-15-22-00-00/
+[............................................................................................................................................................................................................] [100%]
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:209]:
```suggestion

```

Remove output

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 14]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
 ----
 +
-Note that until https://bugzilla.redhat.com/show_bug.cgi?id=1745081[BZ#1745081] is resolved, you must enter the full path `/var/lib/pulp/katello-export/`.
-Relative paths do not work.
+Note you must enter the full path `/var/lib/pulp/imports/<path>`. Relative paths do not work.
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:212]:
```suggestion
Note that you must enter the full path `/var/lib/pulp/imports/<path>`. Relative paths do not work.
```

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 15]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
+---|----------|---------|-------------|-----------------------
+----
+
+
+
+
+[[Using_ISS-Exporting-Library]]
+=== Exporting a Content View Version
+
+You can export contents of all yum repositories in the Library environment of an Organization to an archive file from {ProjectServer} and use this archive file to create the same repositories in another {ProjectServer} or on another {ProjectServer} organization.
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:233]:
```suggestion
You can export contents of all Yum repositories in the Library environment of an organization to an archive file from {ProjectServer} and use this archive file to create the same repositories in another {ProjectServer} or in another {ProjectServer} organization.
```

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 16]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
+The exported archive file contains the following data:
+
+* A JSON file containing Content View version metadata
+* An archive file containing all the repositories from the Library environment of the Organization.
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:237]:
```suggestion
* An archive file containing all the repositories from the Library environment of the organization.
```

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 17]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
+
+{ProjectServer} exports only RPM and kickstart files added to a version of a Content View.
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:239]: What is this trying to say? 

x exports only the RPM and kickstart files that are included in the Content Vew version? 

Or x exports only the RPM and kickstart files to a version of a Content View?

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 18]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
+{Project} does not export the following content:
+
+* Puppet content
+* Docker content
+* OSTree content
+
+.Prerequisites
+
+To export a contents of Library lifecycle envionment of the Organization, ensure that the {ProjectServer} where you want to export meets the following conditions:
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:248]:
```suggestion
To export a contents of Library lifecycle envionment of the organization, ensure that the {ProjectServer} where you want to export meets the following conditions:
```

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 19]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
+
+* Ensure that the export directory has free storage space to accommodate the export.
+* Ensure that the `/var/lib/pulp/exports` directory has free storage space equivalent to the size of the repositories being exported for temporary files created during the export process.
+* Ensure that you set download policy to *Immediate* for all repositories within the Library lifecycle environment you export.
+For more information, see xref:Importing_Red_Hat_Content-Configuring_Download_Policies[].
+* Ensure that you synchronize Products that you export to the required date.
+
+.To Export the Library Content of an Organization:
+
+Get the organization name or id. In the case below we are interested in fully exporting organization ``.
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:258]:
```suggestion
Use the organization name or ID to export. 
```

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 20]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
+
+[options="nowrap" subs="+quotes"]
+----
+# hammer content-export complete library --organization=export-21527
+[............................................................................................................................................................................................................] [100%]
+Generated /var/lib/pulp/exports/export-21527/Export-Library/1.0/2021-03-02T03-35-24-00-00/metadata.json
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:264]:
```suggestion

```

Remove output

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 21]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
+[options="nowrap" subs="+quotes"]
+----
+# ls -lh /var/lib/pulp/exports/export-21527/Export-Library/1.0/2021-03-02T03-35-24-00-00
+total 68M
+-rw-r--r--. 1 pulp pulp 68M Mar  2 03:35 export-1e25417c-6d09-49d4-b9a5-23df4db3d52a-20210302_0335.tar.gz
+-rw-r--r--. 1 pulp pulp 333 Mar  2 03:35 export-1e25417c-6d09-49d4-b9a5-23df4db3d52a-20210302_0335-toc.json
+-rw-r--r--. 1 root root 443 Mar  2 03:35 metadata.json
+----
+
+. You would need all the 3 files ie the `tar.gz` archive file, the `toc.json` and `metadata.json` to be able to import.
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:278]:
```suggestion
. You require all three files, for example, the `tar.gz`, the `toc.json` and the `metadata.json` file to be able to import.
```

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 22]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
+. Notice that a new Content View called `Export-Library` gets created in the Organization. This content view will contain all the repositories belonging to this organization. A new version of this content view is published and exported automatically.
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:279]:
```suggestion
. A new Content View  **Export-Library** is created in the organization. This content view contains all the repositories belonging to this organization. A new version of this Content View is published and exported automatically.
```

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 23]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
+[subs="+quotes"]
+----
+# hammer content-import library --organization=import-32158 \
+                                --path=/var/lib/pulp/imports/2021-03-02T03-35-24-00-00
+[............................................................................................................................................................................................................] [100%]
 ----
++
+Note you must enter the full path `/var/lib/pulp/imports/<path>`. Relative paths do not work.
++
+. To verify that you imported the Libraryr Content successfully check the Contents of the Product and Repositories appropriately.
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:364]:
```suggestion
. To verify that you imported the Library content, check the contents of the Product and Repositories.
```

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc [hunk 24]
```diff
@@ -106,45 +157,242 @@ You can customize the version numbers by changing the `major` and `minor` settin
...
+. Notice that a new content view called `Import-Library` is created in the importing organization. This content view is used to facilitate the library content import.
```
Comment (melcorr) [inline guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:365]:
```suggestion
A new Content View called `Import-Library` is created in the target organization.
This Content View is used to facilitate the library content import.
```

This is not a step.

**Review by melcorr — CHANGES_REQUESTED** [review]
Hi @parthaa 

All of my comments are docs style related.   Only one or two comments. 
Can you please ensure this doc is one line per sentence. This is another standard we follow.

File: guides/doc-Content_Management_Guide/topics/Using_ISS.adoc (hunk 17 above)
Comment (parthaa) [reply guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:218]: More `x exports only the RPM and kickstart files that are included in the Content Vew version`. Should I just change this to
```suggestion
{ProjectServer} exports only RPM and kickstart files included in a Content View  version.
```

Comment (melcorr) [reply guides/doc-Content_Management_Guide/topics/Using_ISS.adoc:218]: LGTM

**Review by melcorr — APPROVED** [review]
This looks good to go.

**Comment by melcorr** [comment]
@parthaa would you mind rebasing and I will merge? Thank you!

**Comment by lzap** [comment]
I am about to branch off 2.4 series, what is the status please? I can hold off the branching process if this is something that should be in 2.4/4.0. Let me know, thanks.

**Pull request merged**